### ISSUE TITLE
Reducing over a zero-length extent panics with index-out-of-bounds instead of returning

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#18]
## Summary

Every reduction over an axis of extent 0 panics with `index out of bounds: the len is 0 but the index is 0`. These are `Result`-returning APIs, so a caller who handles errors correctly still dies. NumPy and PyTorch both answer here: sum of nothing is `0`, max of nothing is an error.

## Reproduction

```rust
use oxmera::Tensor;
Tensor::zeros([0usize, 3]).sum(&[0, 1]).unwrap();
```

```
thread 'main' panicked at oxmera-tensor-0.1.1/src/cpu.rs:130:39:
index out of bounds: the len is 0 but the index is 0
```

Seven of the eight cases I probed panic (CPU backend, `oxmera 0.1.1` and `main` at `fb79b04`):

| expression | result |
|---|---|
| `zeros([0,3]).sum(&[0,1])` | **panic** |
| `zeros([2,0,3]).sum(&[1])` | **panic** |
| `zeros([0,3]).max(&[0,1])` | **panic** |
| `zeros([0,3]).min(&[0,1])` | **panic** |
| `zeros([0,3]).mean(&[0,1])` | **panic** |
| `zeros([2,0]).softmax(1)` | **panic** |
| `zeros([0]).sum(&[0])` | **panic** |
| `zeros([2,0]).argmax(1,false)` | returns `[0, 0]` — see below |

Elementwise ops on empty tensors are fine (`zeros([0,3]) + zeros([0,3])` works, `relu` works), so this is specific to the reduction path.

## Root cause

`crates/oxmera-tensor/src/cpu.rs`, `CpuBackend::reduce`, the odometer inside `reduce_one`:

```rust
let mut acc = op.identity();
let mut coords = vec![0usize; reduced_dims.len()];
let mut off = offset;
loop {
    acc = op.combine(acc, src[off as usize]);   // <-- reads before testing for emptiness
    ...
}
```

It is a do-while: the first element is combined unconditionally, then the odometer decides whether to continue. When any reduced dimension has extent 0 there is no first element, but the read happens anyway against a zero-length storage.

## Expected

Guard the loop and return the identity when the reduced subspace is empty — `Sum` → `0`, `Max` → `-inf`, `Min` → `+inf`, which is what `op.identity()` already provides. `Mean` divides by zero and should either produce `NaN` (NumPy, with a warning) or a typed `InvalidArgument`; picking one deliberately and documenting it seems better than either panicking or silently returning `NaN`.

## Secondary: argmax fabricates an index

`zeros([2,0]).argmax(1, false)` returns `[0, 0]`. There is no element 0 to point at, so this is a made-up answer that will index out of range downstream. PyTorch raises. Worth deciding alongside the above, since it is the same "what does a reduction of nothing mean" question.

## Suggested regression test

A table-driven test over `{sum, max, min, mean, softmax, argmax} × {[0,3], [2,0,3] axis 1, [0]}` asserting each returns rather than panics.
